worker.go

```diff
@@ -44,36 +45,40 @@ type WorkerMetrics struct {
 	// we do not want to key on the metric's Digest here, because those could
 	// collide, and then we'd have to implement a hashtable on top of go maps,
 	// which would be silly
-	counters   map[samplers.MetricKey]*samplers.Counter
-	gauges     map[samplers.MetricKey]*samplers.Gauge
-	histograms map[samplers.MetricKey]*samplers.Histo
-	sets       map[samplers.MetricKey]*samplers.Set
-	timers     map[samplers.MetricKey]*samplers.Histo
+	counters     map[samplers.MetricKey]*samplers.Counter
+	gauges       map[samplers.MetricKey]*samplers.Gauge
+	histograms   map[samplers.MetricKey]*samplers.Histo
+	sets         map[samplers.MetricKey]*samplers.Set
+	timers       map[samplers.MetricKey]*samplers.Histo
+	statusChecks map[samplers.MetricKey]*samplers.StatusCheck
 
 	// this is for counters which are globally aggregated
 	globalCounters map[samplers.MetricKey]*samplers.Counter
 	// and gauges which are global
 	globalGauges map[samplers.MetricKey]*samplers.Gauge
 
 	// these are used for metrics that shouldn't be forwarded
-	localHistograms map[samplers.MetricKey]*samplers.Histo
-	localSets       map[samplers.MetricKey]*samplers.Set
-	localTimers     map[samplers.MetricKey]*samplers.Histo
+	localHistograms   map[samplers.MetricKey]*samplers.Histo
+	localSets         map[samplers.MetricKey]*samplers.Set
+	localTimers       map[samplers.MetricKey]*samplers.Histo
+	localStatusChecks map[samplers.MetricKey]*samplers.StatusCheck
 }
 
 // NewWorkerMetrics initializes a WorkerMetrics struct
 func NewWorkerMetrics() WorkerMetrics {
 	return WorkerMetrics{
-		counters:        make(map[samplers.MetricKey]*samplers.Counter),
-		globalCounters:  make(map[samplers.MetricKey]*samplers.Counter),
-		globalGauges:    make(map[samplers.MetricKey]*samplers.Gauge),
-		gauges:          make(map[samplers.MetricKey]*samplers.Gauge),
-		histograms:      make(map[samplers.MetricKey]*samplers.Histo),
-		sets:            make(map[samplers.MetricKey]*samplers.Set),
-		timers:          make(map[samplers.MetricKey]*samplers.Histo),
-		localHistograms: make(map[samplers.MetricKey]*samplers.Histo),
-		localSets:       make(map[samplers.MetricKey]*samplers.Set),
-		localTimers:     make(map[samplers.MetricKey]*samplers.Histo),
+		counters:          make(map[samplers.MetricKey]*samplers.Counter),
```

Comment:
`make` is only used when you're initializing a capacity, which you're not doing here. Instead, just use the literal value, for example: `map[samplers.MetricKey]samplers.Counter{}`.

Reply:
This is all copy paste. I'll clean it up.